### DISCUSSION TITLE
Topics for python newbie session!

### DISCUSSION THREAD

**Discussion by sarambl** [pangeo-data/escience-2022#22] [Ideas]
Hi! We said we would do a python newbie session tomorrow and it would be great if you could contribute a couple of topics you would want to discuss :) 
I have in mind: 
1. The Jupyter environment
2. Basic packages like numpy, matplotlib, pandas etc (suggest more?)

**Comment by lovisaandreanilsson** [comment]
Sounds good Sara! 
I struggle with pandas and xarrays. Slicing, accessing data within those and visualization is... challenging haha.

**Comment by MichaelSchulzMETNO** [comment]
I wonder if we should also point to some ways to get help on python questions - a very basic example of nice overview on python syntax and language : https://www.w3schools.com/python/default.asp

**Reply by sarambl** [reply]
Yes! Good tip. Also, I rather like this one, which goes from pretty basic to more advanced: https://jakevdp.github.io/PythonDataScienceHandbook/

**Comment by MichaelSchulzMETNO** [comment]
Possible Topic: How to see/understand methods, arguments in using classes/objects/functions. How to use contextual help in jupyter notebook...

**Comment by sarambl** [comment]
Maybe we could find some problem that we can go through together? Like some analysis? Any suggestions?

**Reply by sarambl** [reply]
Another topic: organising your code and your data

**Reply by theodorekhadir** [reply]
Maybe something using basic python stuff instead of fancy libraries that do everything for you like xarray.
Solve a problem with for loops, also with list of objects (looping in a list is the thing that I use the most in python and not everyone knows about it, could be useful):
- For example calculate the trend/or correlation in every grid cell of map using a for loop and putting the result in a new dataset to be able to plot it (you can do everything without for loops here but sometimes you can't so good to learn).

**Reply by theodorekhadir** [reply]
- enumerate
- dictionaries
- zip
- try except
- flatten
- reshape
- break
